### PR TITLE
review for main function and trivial modification of lib/test files

### PR DESCRIPTION
**Pull request by liuhanyao98**

### CODE REVIEW

File: modeling_tool.cc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <chrono>
+#include <iostream>
+#include <stdexcept>
+
+#include "google/cloud/spanner/client.h"
+
+
+static const std::int64_t BATCHSIZE = 1000; // mutations per commit 
+
+int main(int argc, char* argv[]) try {
```
Comment (zhiwen-fang) [inline modeling_tool.cc]: Don't use exception, use status instead.

File: modeling_tool.cc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  namespace spanner = ::google::cloud::spanner;
+
+  spanner::Client readClient(
+      spanner::MakeConnection(spanner::Database(argv[1], argv[2], argv[3])));
+  spanner::Client writeClient(
+      spanner::MakeConnection(spanner::Database(argv[1], argv[2], argv[3])));
+  
+  auto start = std::chrono::high_resolution_clock::now();
+  
+  const auto& updatedRecord = modeling_tool::batchUpdateData(readClient, writeClient, BATCHSIZE);
```
Comment (zhiwen-fang) [inline modeling_tool.cc]: How can we dry_run?

File: modeling_tool.cc [hunk 3]
```diff
@@ -22,12 +22,16 @@
...
+  if(argc < 4) {
+    std::cerr << "Usage 1: " << argv[0]
               << " project-id instance-id database-id\n";
+    std::cerr << "Usage 2: " << argv[0]
+              << " project-id instance-id database-id --dry-run \n";              
     return 1;
   }
+  bool dryRun = false;
+  if(argc == 5) dryRun = true;
   namespace spanner = ::google::cloud::spanner;
```
Comment (zhiwen-fang) [inline modeling_tool.cc:35]: move it out of main?

File: modeling_tool.h [hunk 4]
```diff
@@ -70,28 +72,35 @@ StatusOr<std::int64_t> batchUpdateData(spanner::Client& readClient,
...
-    if (!commitResult) return commitResult.status();
+    if(!dryRun) {
+      const auto& commitResult = writeClient.Commit(mutations);
+      if (!commitResult) return commitResult.status();
+    }
     updatedRecord += mutations.size();
   }
-  return StatusOr<std::int64_t>(updatedRecord);
+  std::pair<std::int64_t, std::int64_t> stats(readRecord, updatedRecord);
+  return StatusOr<std::pair<std::int64_t, std::int64_t>>(stats);
```
Comment (zhiwen-fang) [inline modeling_tool.h:93]: does the following work?
return stats;

**Comment by zhiwen-fang** [comment]
LGTM

File: modeling_tool.h (hunk 4 above)
Comment (liuhanyao98) [reply modeling_tool.h:93]: yes, done

**Pull request merged**